feat: beatutify busTooltip & make lineProfilePage actually work

itsoriki opened the pull request:
Resolves #543 

# Description
- make LineProfilePage **fully functional**
- style `BusTooltip` according to language (also add a collapsible section)

## screenshots
![image](https://github.com/hasadna/open-bus-map-search/assets/93577239/3603e05c-820b-4944-bbd2-0557a7ad25ea)
![image](https://github.com/hasadna/open-bus-map-search/assets/93577239/24f04e28-917a-4462-ab45-4786cc265f6d)
![image](https://github.com/hasadna/open-bus-map-search/assets/93577239/8c2aec6c-0e90-483c-8618-f6d58724a9dc)
![image](https://github.com/hasadna/open-bus-map-search/assets/93577239/ff672ed6-03bf-4cb8-9719-4342cc833bde)
![image](https://github.com/hasadna/open-bus-map-search/assets/93577239/5786b714-3468-4476-9a8d-16b96932dc16)
![image](https://github.com/hasadna/open-bus-map-search/assets/93577239/bb9b2184-3e25-4637-9932-97f5cd21afac)
![image](https://github.com/hasadna/open-bus-map-search/assets/93577239/3df86cff-a797-433c-bdaf-0ba062d6e273)

itsoriki commented:
https://github.com/all-contributors please add @itsoriki as code contributor

itsoriki commented:
@NoamGaash I can't figure out why that one job is failing consistently - No real error is printed to the console...
![image](https://github.com/hasadna/open-bus-map-search/assets/93577239/c187c9f1-d2ec-4875-9da3-8e3bc58feb2f)

NoamGaash commented:
@itsoriki yup that's bad
let's start with #747

NoamGaash commented:
run `npm run lint` on your branch

itsoriki commented:
@NoamGaash Sorry for the "mess" - can you review this PR now that everything has passed?

NoamGaash commented on src/pages/components/map-related/MapLayers/BusToolTip.tsx:
```diff
@@ -53,17 +57,20 @@ export function BusToolTip({ position, icon }: BusToolTipProps) {
 
             <li>
-              {t('sample_time')} :
+              {`${t('sample_time')}: `}
```
why not
```suggestion
              {t('sample_time')}:
```
?

itsoriki replied:
The reason being the single space at the end of the string - doing it like you suggest will leave no space between the text and the data (unless we'll add an ugly `{' '}`) so I figured this is the cleanest approach to achieve the goal here

NoamGaash commented on src/pages/lineProfile/LineProfile.tsx:
```diff
@@ -0,0 +1,132 @@
+  const [availableRoutes, setAvailableRoutes] = useState<BusRoute[] | null>(null)
+  const [selectedRouteKey, setSelectedRouteKey] = useState<string>('')
+
+  useEffect(() => {
+    window.scrollTo(0, 0)
+  }, [])
```
why is it needed?

itsoriki replied:
I don't know why, but when you first arrive on this page, you are not at the top of it (i.e. you don't see the title for example) - see picture

![image](https://github.com/hasadna/open-bus-map-search/assets/93577239/ba2bf1f5-25cb-4c63-93f4-de0a3c5a22a4)

Is there a better way to do it?

NoamGaash replied:
I'm not sure, we can investigate that later

NoamGaash commented on src/pages/lineProfile/LineProfile.tsx:
```diff
@@ -0,0 +1,132 @@
+  )
+}
+
+const LineProfileMapContainer = styled.div`
+  .map-info {
+    height: 15rem;
+  }
+`
```
should we move it to the scss file?

itsoriki replied:
Done

NoamGaash reviewed (approved):
That's a great improvement!
few suggestions:
1. I'm not sure that we need the expanding button - what is it good for?
2. We can consider translating the direction (290 מעלות) to geographic terms (צפון מערב, אזימוט 290)
3. when the tooltip is too tall. we can add an inner scrollbar or something similar

wdyt?

itsoriki commented:
> That's a great improvement! few suggestions:
> 
> 1. I'm not sure that we need the expanding button - what is it good for?
> 2. We can consider translating the direction (290 מעלות) to geographic terms (צפון מערב, אזימוט 290)
> 3. when the tooltip is too tall. we can add an inner scrollbar or something similar
> 
> wdyt?

1. Removed the expanding button
2. I'll open a new (good-first) issue for that
3. Added an inner scrollbar to the `BusTooltip`

NoamGaash commented:
Thank you! :clap: 

@all-contributor please add @itsoriki for code :)

NoamGaash commented:
maybe
@all-contributors please add @itsoriki for code 
?
